OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -93,15 +93,16 @@ public class Armament : UpgradableTrait<ArmamentInfo>, ITick, IExplodeModifier
 		public readonly Barrel[] Barrels;
 
 		readonly Actor self;
-		Lazy<Turreted> turret;
+		Turreted turret;
+		AmmoPool ammoPool;
 		Lazy<BodyOrientation> coords;
-		Lazy<AmmoPool> ammoPool;
 		List<Pair<int, Action>> delayedActions = new List<Pair<int, Action>>();
 		Lazy<IEnumerable<int>> rangeModifiers;
```

Comment:
It won't.

The reason they're lazy is because the traits they're looking up may not exist in the constructor (trait initialization order and whatnot) - so by making them lazy we can avoid doing the lookup until later when they actually exist.

As long as you do the lookup in `INotifyCreated` then you'll be fine - as all the traits will have been fully constructed by that point in time.